bitnami/shared/kyverno/kafka-client-policies.yaml

```diff
@@ -0,0 +1,117 @@
+            duration: 168h0m0s # 7 days, renews 2/3 of the way through
+            issuerRef:
+              group: awspca.cert-manager.io
+              kind: AWSPCAClusterIssuer
+              name: aws-pca
+            secretName: "{{request.object.metadata.name}}-gen-kafka-cert"
+    - name: generate-uw-hosted-certificate
+      match:
+        any:
+          - resources:
+              annotations:
+                uw.systems/uw-hosted-kafka-client: "*"
+              kinds:
+                - Deployment
+                - StatefulSet
+      generate:
+        # Delete certificate if the target is deleted
+        synchronize: true
+        # Keep the generated certificates if the policy is deleted
+        orphanDownstreamOnPolicyDelete: true
+        apiVersion: cert-manager.io/v1
+        kind: Certificate
+        name: "{{request.object.metadata.name}}-gen-kafka-cert"
+        namespace: "{{request.namespace}}"
+        data:
+          spec:
+            commonName: "{{request.namespace}}/{{request.object.metadata.annotations.\"uw.systems/uw-hosted-kafka-client\"}}"
+            duration: 168h0m0s # 7 days, renews 2/3 of the way through
+            issuerRef:
+              kind: ClusterIssuer
+              name: kafka-shared-selfsigned-issuer
+            secretName: "{{request.object.metadata.name}}-gen-kafka-cert"
+---
+
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: mount-kafka-certificate-volume
+  annotations:
+    policies.kyverno.io/title: Add Certificates as a Volume
+    policies.kyverno.io/category: Sample
+    policies.kyverno.io/subject: Deployment,Volume
+    policies.kyverno.io/description: >-
+      This policy mounts the TLS artifacts from a certificate in all the containers of the annotated subjects.
+spec:
+  background: false
+  rules:
+    - name: add-ssl-certs
+      match:
+        any:
+          - resources:
+              annotations:
+                # this will match all annotations for kafka client certificate
+                uw.systems/*-kafka-client: "*"
+              kinds:
+                - Deployment
+                - StatefulSet
+              operations:
+                - CREATE
+                - UPDATE
+      mutate:
+        # mount the certificates volume on all containers in the pod, as we don't know which is the "app" one.
+        foreach:
+          - list: "request.object.spec.template.spec.containers"
+            patchStrategicMerge:
+              spec:
+                template:
+                  spec:
+                    containers:
+                      - name: "{{ element.name }}"
+                        env:
+                          - name: UW_KAFKA_TLS_AUTH
+                            value: "true"
+                        volumeMounts:
+                        - name: kafka-client-cert
+                          mountPath: /certs
+                          readOnly: true
+                    volumes:
```

Comment:
I think volumes should not be added for each of containers. 
Not sure, how will it work with PatchStrategicMerge

Reply:
`volumeMounts` should added per container, but there should be only one `volume` per resource.
If you are adding it per container, it means you are overwriting it as many times as there is containers, right? That isn't best approach, but final result is okay. 
I'm just worried if you are not adding the same `volume` multiple times this way.

Reply:
Yea, indeed, I saw that. I followed the example from here: https://artifacthub.io/packages/kyverno/kyverno-policies/add-certificates-volume
Seems to work like this.